### DISCUSSION TITLE
How to import class outside namespace?

### DISCUSSION THREAD

**Discussion by cross-hello** [pythonnet/pythonnet#1699] [Q&A] [answered]
### Environment

-   Pythonnet version:  2.5.2
-   Python version: 3.8.10
-   Operating System:Windows 10
-   .NET Runtime: 4.7.2

### Details

-   Import class without namespace, just like following:
```C#
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Linq;
using System.Collections.Generic;

public class t
{
    public void p()
    {
        Console.WriteLine("Welcome come to here");
    }
}
```
 How could I import **t** class?

**Comment by cross-hello** [comment, marked as answer]
Ok, thanks God. I got out the solution:
```Python
>>> import clr
>>> clr.AddReference(r'bin/Debug/ClassLibrary1')
>>>dir(clr)
['AddReference', 'ClassLibrary1', 'ClrLoader', 'FindAssembly', 'GetClrType', 'ListAssemblies', 'Microsoft', 'Python', 'System', '__class__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', '__version__', '_add_pending_namespaces', '_available_namespaces', '_extras', '_load_clr_module', 'clrmethod', 'clrproperty', 'getPreload', 'loader', 'setPreload', 't']
```
Now you could see that **t** is listed inside clr element. So you could reference **t** in global namespace as this: `clr.t`
That is all. Thank you.

**Reply by ross-g** [reply]
Just wanted to say thanks for posting the answer to your question. This was not something I would ever have thought to try.

**Comment by cross-hello** [comment]
Thanks God.

May 25, 2022 19:42:31 Ross ***@***.***>:
